src/model/message/posted-item.js

```diff
@@ -83,8 +84,18 @@ export class PostedItemMessageModel extends WarhammerMessageModel {
     let paid = await game.wfrp4e.market.handlePlayerPayment({payString : game.wfrp4e.market.amountToString(this.itemData.system.price), itemData : this.itemData});
     for(let actor of paid)
     {
-      await actor.createEmbeddedDocuments("Item", [this.itemData], {fromMessage: this.parent.id})
-      ui.notifications.notify(game.i18n.format("MARKET.ItemAdded", { item: this.itemData.name, actor : actor.name })) 
+      const hasItem = actor.itemTypes[this.itemData.type].find(i => i.name === this.itemData.name
+          && i._stats.compendiumSource === this.itemData._stats.compendiumSource);
+      if(hasItem) {
+        await actor.updateEmbeddedDocuments("Item", [{
+          _id : hasItem.id,
+          "system.quantity.value" : hasItem.system.quantity.value + this.itemData.system.quantity.value,
+        }], {fromMessage: this.parent.id});
+        ui.notifications.notify(game.i18n.format("MARKET.ItemAppended", { item: this.itemData.name, actor : actor.name, quantity: hasItem.system.quantity.value }));
+      } else {
+        await actor.createEmbeddedDocuments("Item", [this.itemData], {fromMessage: this.parent.id});
+        ui.notifications.notify(game.i18n.format("MARKET.ItemAdded", { item: this.itemData.name, actor : actor.name }));
```

Comment:
You can add format to the second argument, like

`ui.notifications.info("key", {format : {...yourArguments}});`